### PR TITLE
Fixing flatpak

### PR DESCRIPTION
**Pull request by albfan**
Taking flathub as example I fixed flatpak compilation to check under an elementary sandbox

I did install org.gnome.Sdk.Debug and get backtraces like:

<details>
<summary>backtrace</summary>

```
Thread 1 "com.github.alec" received signal SIGSEGV, Segmentation fault.
0x0000000000000000 in ?? ()
(gdb) bt
#0  0x0000000000000000 in ?? ()
#1  0x00007ffff784c470 in gtk_css_value_border_compute (value=0x5555556ce580, property_id=58, provider=0x7fffec003990, style=0x555555f70370, parent_style=0x555555f73480)
    at gtkcssbordervalue.c:60
#2  0x00007ffff7874938 in gtk_css_static_style_compute_value (style=0x555555f70370, provider=0x7fffec003990, parent_style=0x555555f73480, id=58, 
    specified=0x7ffff7e16b30 <initial>, section=0x0) at gtkcssstaticstyle.c:237
#3  0x00007ffff785fc82 in _gtk_css_lookup_resolve (lookup=lookup@entry=0x555555ed9000, provider=provider@entry=0x7fffec003990, style=style@entry=0x555555f70370, 
    parent_style=parent_style@entry=0x555555f73480) at gtkcsslookup.c:122
#4  0x00007ffff7874860 in gtk_css_static_style_new_compute (provider=0x7fffec003990, matcher=matcher@entry=0x7fffffffb190, parent=parent@entry=0x555555f73480)
    at gtkcssstaticstyle.c:195
#5  0x00007ffff78622e5 in gtk_css_node_create_style (cssnode=0x7fffdc013e10) at gtkcssnode.c:371
#6  gtk_css_node_real_update_style (cssnode=0x7fffdc013e10, change=64424509425, timestamp=41135048752, style=0x555555f745b0) at gtkcssnode.c:425
#7  0x00007ffff7861094 in gtk_css_node_ensure_style (cssnode=cssnode@entry=0x7fffdc013e10, current_time=41135048752) at gtkcssnode.c:1007
#8  0x00007ffff786127e in gtk_css_node_ensure_style (current_time=<optimized out>, cssnode=0x7fffdc013e10) at gtkcssnode.c:1033
#9  gtk_css_node_get_style (cssnode=0x7fffdc013e10) at gtkcssnode.c:1033
#10 0x00007ffff7853059 in gtk_css_gadget_get_style (gadget=0x7fffdc013f70) at gtkcssgadget.c:645
#11 gtk_css_gadget_get_preferred_size (gadget=0x7fffdc013f70, orientation=orientation@entry=GTK_ORIENTATION_VERTICAL, for_size=for_size@entry=-1, 
    minimum=minimum@entry=0x7fffffffb3b8, natural=natural@entry=0x7fffffffb3bc, minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtkcssgadget.c:645
#12 0x00007ffff7975853 in gtk_range_size_request (widget=0x555555e59170, orientation=GTK_ORIENTATION_VERTICAL, minimum=0x7fffffffb3b8, natural=0x7fffffffb3bc)
    at gtkrange.c:1909
#13 0x00007ffff79b2fcb in gtk_widget_query_size_for_orientation (widget=widget@entry=0x555555e59170, orientation=orientation@entry=GTK_ORIENTATION_VERTICAL, 
    for_size=for_size@entry=-1, minimum_size=minimum_size@entry=0x7fffffffb540, natural_size=natural_size@entry=0x7fffffffb544, 
    minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtksizerequest.c:219
#14 0x00007ffff79b3730 in gtk_widget_compute_size_for_orientation (widget=widget@entry=0x555555e59170, orientation=orientation@entry=GTK_ORIENTATION_VERTICAL, 
    for_size=for_size@entry=-1, minimum=minimum@entry=0x7fffffffb540, natural=natural@entry=0x7fffffffb544, minimum_baseline=minimum_baseline@entry=0x0, 
    natural_baseline=0x0) at gtksizerequest.c:399
#15 0x00007ffff79b3a65 in gtk_widget_get_preferred_height_and_baseline_for_width (widget=widget@entry=0x555555e59170, width=width@entry=-1, 
    minimum_height=minimum_height@entry=0x7fffffffb540, natural_height=natural_height@entry=0x7fffffffb544, minimum_baseline=minimum_baseline@entry=0x0, 
    natural_baseline=natural_baseline@entry=0x0) at gtksizerequest.c:642
#16 0x00007ffff79b3bab in _gtk_widget_get_preferred_size_and_baseline (widget=0x555555e59170, minimum_size=minimum_size@entry=0x7fffffffb598, 
    natural_size=natural_size@entry=0x0, minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=natural_baseline@entry=0x0) at gtksizerequest.c:703
#17 0x00007ffff79b3bfe in gtk_widget_get_preferred_size (widget=<optimized out>, minimum_size=minimum_size@entry=0x7fffffffb598, natural_size=natural_size@entry=0x0)
    at gtksizerequest.c:750
#18 0x00007ffff799d049 in gtk_scrolled_window_measure (gadget=<optimized out>, orientation=GTK_ORIENTATION_HORIZONTAL, for_size=<optimized out>, 
    minimum_size=0x7fffffffb738, natural_size=0x7fffffffb73c, minimum_baseline=<optimized out>, natural_baseline=0x0, data=0x0) at gtkscrolledwindow.c:1835
#19 0x00007ffff784ede7 in gtk_css_custom_gadget_get_preferred_size (gadget=<optimized out>, orientation=<optimized out>, for_size=<optimized out>, 
    minimum=<optimized out>, natural=<optimized out>, minimum_baseline=<optimized out>, natural_baseline=0x0) at gtkcsscustomgadget.c:124
#20 0x00007ffff7853179 in gtk_css_gadget_get_preferred_size (gadget=0x7fffdc013b70, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum=0x7fffffffb738, natural=0x7fffffffb73c, minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtkcssgadget.c:683
#21 0x00007ffff79992e7 in gtk_scrolled_window_get_preferred_width (widget=<optimized out>, minimum_size=<optimized out>, natural_size=<optimized out>)
    at gtkscrolledwindow.c:4092
#22 0x00007ffff79b2f01 in gtk_widget_query_size_for_orientation (widget=widget@entry=0x7fffdc0184e0, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
--Type <RET> for more, q to quit, c to continue without paging--
    for_size=for_size@entry=-1, minimum_size=minimum_size@entry=0x7fffffffba38, natural_size=natural_size@entry=0x7fffffffba3c, 
    minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtksizerequest.c:181
#23 0x00007ffff79b3730 in gtk_widget_compute_size_for_orientation (widget=widget@entry=0x7fffdc0184e0, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum=minimum@entry=0x7fffffffba38, natural=natural@entry=0x7fffffffba3c, minimum_baseline=minimum_baseline@entry=0x0, 
    natural_baseline=0x0) at gtksizerequest.c:399
#24 0x00007ffff79b381e in gtk_widget_get_preferred_width (widget=widget@entry=0x7fffdc0184e0, minimum_width=minimum_width@entry=0x7fffffffba38, 
    natural_width=natural_width@entry=0x7fffffffba3c) at gtksizerequest.c:492
#25 0x00007ffff79b3e70 in _gtk_widget_get_preferred_size_for_size (widget=0x7fffdc0184e0, orientation=GTK_ORIENTATION_HORIZONTAL, size=-1, minimum=0x7fffffffba38, 
    natural=0x7fffffffba3c, minimum_baseline=0x0, natural_baseline=0x0) at gtksizerequest.c:871
#26 0x00007ffff784ede7 in gtk_css_custom_gadget_get_preferred_size (gadget=<optimized out>, orientation=<optimized out>, for_size=<optimized out>, 
    minimum=<optimized out>, natural=<optimized out>, minimum_baseline=<optimized out>, natural_baseline=0x0) at gtkcsscustomgadget.c:124
#27 0x00007ffff7853179 in gtk_css_gadget_get_preferred_size (gadget=0x7fffdc025ba0, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum=0x7fffffffba38, natural=0x7fffffffba3c, minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtkcssgadget.c:683
#28 0x00007ffff7a57637 in gtk_viewport_get_preferred_width (widget=<optimized out>, minimum_size=<optimized out>, natural_size=<optimized out>) at gtkviewport.c:1070
#29 0x00007ffff79b2f01 in gtk_widget_query_size_for_orientation (widget=widget@entry=0x555555ee8c10, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum_size=minimum_size@entry=0x7fffffffbba8, natural_size=natural_size@entry=0x7fffffffbbac, 
    minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtksizerequest.c:181
#30 0x00007ffff79b3730 in gtk_widget_compute_size_for_orientation (widget=widget@entry=0x555555ee8c10, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum=minimum@entry=0x7fffffffbba8, natural=natural@entry=0x7fffffffbbac, minimum_baseline=minimum_baseline@entry=0x0, 
    natural_baseline=0x0) at gtksizerequest.c:399
#31 0x00007ffff79b381e in gtk_widget_get_preferred_width (widget=widget@entry=0x555555ee8c10, minimum_width=minimum_width@entry=0x7fffffffbba8, 
    natural_width=natural_width@entry=0x7fffffffbbac) at gtksizerequest.c:492
#32 0x00007ffff799d0fa in gtk_scrolled_window_measure (gadget=<optimized out>, orientation=GTK_ORIENTATION_HORIZONTAL, for_size=<optimized out>, 
    minimum_size=0x7fffffffbd58, natural_size=0x7fffffffbd5c, minimum_baseline=<optimized out>, natural_baseline=0x0, data=0x0) at gtkscrolledwindow.c:1850
#33 0x00007ffff784ede7 in gtk_css_custom_gadget_get_preferred_size (gadget=<optimized out>, orientation=<optimized out>, for_size=<optimized out>, 
    minimum=<optimized out>, natural=<optimized out>, minimum_baseline=<optimized out>, natural_baseline=0x0) at gtkcsscustomgadget.c:124
#34 0x00007ffff7853179 in gtk_css_gadget_get_preferred_size (gadget=0x55555595e060, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum=0x7fffffffbd58, natural=0x7fffffffbd5c, minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtkcssgadget.c:683
#35 0x00007ffff79992e7 in gtk_scrolled_window_get_preferred_width (widget=<optimized out>, minimum_size=<optimized out>, natural_size=<optimized out>)
    at gtkscrolledwindow.c:4092
#36 0x00007ffff79b2f01 in gtk_widget_query_size_for_orientation (widget=widget@entry=0x555555912780, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum_size=minimum_size@entry=0x7fffffffbf08, natural_size=natural_size@entry=0x7fffffffbf10, 
    minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtksizerequest.c:181
#37 0x00007ffff79b3730 in gtk_widget_compute_size_for_orientation (widget=widget@entry=0x555555912780, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum=minimum@entry=0x7fffffffbf08, natural=natural@entry=0x7fffffffbf10, minimum_baseline=minimum_baseline@entry=0x0, 
    natural_baseline=0x0) at gtksizerequest.c:399
#38 0x00007ffff79b381e in gtk_widget_get_preferred_width (widget=0x555555912780, minimum_width=minimum_width@entry=0x7fffffffbf08, 
    natural_width=natural_width@entry=0x7fffffffbf10) at gtksizerequest.c:492
#39 0x00007ffff78d848b in compute_request_for_child (natural_baseline=0x7fffffffbf14, minimum_baseline=0x7fffffffbf0c, natural=0x7fffffffbf10, minimum=0x7fffffffbf08, 
    contextual=0, orientation=GTK_ORIENTATION_HORIZONTAL, child=<optimized out>, request=0x7fffffffbfa0) at gtkgrid.c:681
#40 gtk_grid_request_non_spanning (contextual=0, orientation=GTK_ORIENTATION_HORIZONTAL, request=0x7fffffffbfa0) at gtkgrid.c:723
--Type <RET> for more, q to quit, c to continue without paging--
#41 gtk_grid_request_run (request=request@entry=0x7fffffffbfa0, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, contextual=contextual@entry=0)
    at gtkgrid.c:1132
#42 0x00007ffff78d86d5 in gtk_grid_get_size (grid=<optimized out>, orientation=GTK_ORIENTATION_HORIZONTAL, minimum=0x7fffffffc158, natural=0x7fffffffc15c, 
    minimum_baseline=0x0, natural_baseline=<optimized out>) at gtkgrid.c:1438
#43 0x00007ffff784ede7 in gtk_css_custom_gadget_get_preferred_size (gadget=<optimized out>, orientation=<optimized out>, for_size=<optimized out>, 
    minimum=<optimized out>, natural=<optimized out>, minimum_baseline=<optimized out>, natural_baseline=0x0) at gtkcsscustomgadget.c:124
#44 0x00007ffff7853179 in gtk_css_gadget_get_preferred_size (gadget=0x555555871870, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum=0x7fffffffc158, natural=0x7fffffffc15c, minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtkcssgadget.c:683
#45 0x00007ffff78d6747 in gtk_grid_get_preferred_width (widget=<optimized out>, minimum=<optimized out>, natural=<optimized out>) at gtkgrid.c:1490
#46 0x00007ffff79b2f01 in gtk_widget_query_size_for_orientation (widget=widget@entry=0x555555926610, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum_size=minimum_size@entry=0x7fffffffc2e0, natural_size=natural_size@entry=0x7fffffffc2e4, 
    minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtksizerequest.c:181
#47 0x00007ffff79b3730 in gtk_widget_compute_size_for_orientation (widget=widget@entry=0x555555926610, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum=minimum@entry=0x7fffffffc2e0, natural=natural@entry=0x7fffffffc2e4, minimum_baseline=minimum_baseline@entry=0x0, 
    natural_baseline=0x0) at gtksizerequest.c:399
#48 0x00007ffff79b381e in gtk_widget_get_preferred_width (widget=widget@entry=0x555555926610, minimum_width=minimum_width@entry=0x7fffffffc2e0, 
    natural_width=natural_width@entry=0x7fffffffc2e4) at gtksizerequest.c:492
#49 0x00007ffff79b9f0a in gtk_stack_measure (gadget=<optimized out>, orientation=GTK_ORIENTATION_HORIZONTAL, for_size=-1, minimum=0x7fffffffc478, 
    natural=0x7fffffffc47c, minimum_baseline=<optimized out>, natural_baseline=0x0, data=0x0) at gtkstack.c:2410
#50 0x00007ffff784ede7 in gtk_css_custom_gadget_get_preferred_size (gadget=<optimized out>, orientation=<optimized out>, for_size=<optimized out>, 
    minimum=<optimized out>, natural=<optimized out>, minimum_baseline=<optimized out>, natural_baseline=0x0) at gtkcsscustomgadget.c:124
#51 0x00007ffff7853179 in gtk_css_gadget_get_preferred_size (gadget=0x55555574c500, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum=0x7fffffffc478, natural=0x7fffffffc47c, minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtkcssgadget.c:683
#52 0x00007ffff79bac2b in gtk_stack_get_preferred_width (widget=<optimized out>, minimum=<optimized out>, natural=<optimized out>) at gtkstack.c:2311
#53 0x00007ffff79b2f01 in gtk_widget_query_size_for_orientation (widget=widget@entry=0x555555653450, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum_size=minimum_size@entry=0x7fffffffc64c, natural_size=natural_size@entry=0x7fffffffc650, 
    minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtksizerequest.c:181
#54 0x00007ffff79b3730 in gtk_widget_compute_size_for_orientation (widget=widget@entry=0x555555653450, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum=minimum@entry=0x7fffffffc64c, natural=natural@entry=0x7fffffffc650, minimum_baseline=minimum_baseline@entry=0x0, 
    natural_baseline=0x0) at gtksizerequest.c:399
#55 0x00007ffff79b381e in gtk_widget_get_preferred_width (widget=widget@entry=0x555555653450, minimum_width=minimum_width@entry=0x7fffffffc64c, 
    natural_width=natural_width@entry=0x7fffffffc650) at gtksizerequest.c:492
#56 0x00007ffff79b3e70 in _gtk_widget_get_preferred_size_for_size (widget=0x555555653450, orientation=GTK_ORIENTATION_HORIZONTAL, size=size@entry=-1, 
    minimum=minimum@entry=0x7fffffffc64c, natural=natural@entry=0x7fffffffc650, minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0)
    at gtksizerequest.c:871
#57 0x00007ffff794c1c2 in gtk_paned_get_preferred_size_for_orientation (natural=0x7fffffffc7ec, minimum=0x7fffffffc7e8, size=-1, widget=<optimized out>)
    at gtkpaned.c:1076
#58 gtk_paned_measure (gadget=<optimized out>, orientation=GTK_ORIENTATION_HORIZONTAL, size=-1, minimum=0x7fffffffc7e8, natural=0x7fffffffc7ec, 
    minimum_baseline=<optimized out>, natural_baseline=0x0, data=0x0) at gtkpaned.c:1237
#59 0x00007ffff784ede7 in gtk_css_custom_gadget_get_preferred_size (gadget=<optimized out>, orientation=<optimized out>, for_size=<optimized out>, 
    minimum=<optimized out>, natural=<optimized out>, minimum_baseline=<optimized out>, natural_baseline=0x0) at gtkcsscustomgadget.c:124
--Type <RET> for more, q to quit, c to continue without paging--
#60 0x00007ffff7853179 in gtk_css_gadget_get_preferred_size (gadget=0x55555574c280, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum=0x7fffffffc7e8, natural=0x7fffffffc7ec, minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtkcssgadget.c:683
#61 0x00007ffff794b687 in gtk_paned_get_preferred_width (widget=<optimized out>, minimum=<optimized out>, natural=<optimized out>) at gtkpaned.c:1247
#62 0x00007ffff79b2f01 in gtk_widget_query_size_for_orientation (widget=widget@entry=0x55555565c620, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum_size=minimum_size@entry=0x7fffffffc968, natural_size=natural_size@entry=0x7fffffffc96c, 
    minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtksizerequest.c:181
#63 0x00007ffff79b3730 in gtk_widget_compute_size_for_orientation (widget=widget@entry=0x55555565c620, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum=minimum@entry=0x7fffffffc968, natural=natural@entry=0x7fffffffc96c, minimum_baseline=minimum_baseline@entry=0x0, 
    natural_baseline=0x0) at gtksizerequest.c:399
#64 0x00007ffff79b381e in gtk_widget_get_preferred_width (widget=widget@entry=0x55555565c620, minimum_width=minimum_width@entry=0x7fffffffc968, 
    natural_width=natural_width@entry=0x7fffffffc96c) at gtksizerequest.c:492
#65 0x00007ffff7a74192 in gtk_window_get_preferred_width (widget=0x5555559062e0, minimum_size=0x7fffffffca78, natural_size=0x7fffffffca7c) at gtkwindow.c:8782
#66 0x00007ffff77efa64 in gtk_application_window_real_get_preferred_width (widget=0x5555559062e0, minimum_width=0x7fffffffca78, natural_width=0x7fffffffca7c)
    at gtkapplicationwindow.c:576
#67 0x00007ffff79b2f01 in gtk_widget_query_size_for_orientation (widget=widget@entry=0x5555559062e0, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum_size=minimum_size@entry=0x7fffffffcbe0, natural_size=natural_size@entry=0x7fffffffcbe4, 
    minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=0x0) at gtksizerequest.c:181
#68 0x00007ffff79b3730 in gtk_widget_compute_size_for_orientation (widget=widget@entry=0x5555559062e0, orientation=orientation@entry=GTK_ORIENTATION_HORIZONTAL, 
    for_size=for_size@entry=-1, minimum=minimum@entry=0x7fffffffcbe0, natural=natural@entry=0x7fffffffcbe4, minimum_baseline=minimum_baseline@entry=0x0, 
    natural_baseline=0x0) at gtksizerequest.c:399
#69 0x00007ffff79b381e in gtk_widget_get_preferred_width (widget=widget@entry=0x5555559062e0, minimum_width=minimum_width@entry=0x7fffffffcbe0, 
    natural_width=natural_width@entry=0x7fffffffcbe4) at gtksizerequest.c:492
#70 0x00007ffff79b3b4c in _gtk_widget_get_preferred_size_and_baseline (widget=widget@entry=0x5555559062e0, minimum_size=minimum_size@entry=0x7fffffffcc80, 
    natural_size=natural_size@entry=0x0, minimum_baseline=minimum_baseline@entry=0x0, natural_baseline=natural_baseline@entry=0x0) at gtksizerequest.c:685
#71 0x00007ffff79b3bfe in gtk_widget_get_preferred_size (widget=widget@entry=0x5555559062e0, minimum_size=minimum_size@entry=0x7fffffffcc80, 
    natural_size=natural_size@entry=0x0) at gtksizerequest.c:750
#72 0x00007ffff7a75714 in gtk_window_compute_hints (new_flags=<synthetic pointer>, new_geometry=0x7fffffffcc90, window=0x5555559062e0) at gtkwindow.c:10234
#73 gtk_window_compute_configure_request (window=window@entry=0x5555559062e0, request=request@entry=0x7fffffffcd80, geometry=geometry@entry=0x7fffffffcda0, 
    flags=flags@entry=0x7fffffffcd78) at gtkwindow.c:9550
#74 0x00007ffff7a7611c in gtk_window_move_resize (window=0x5555559062e0) at gtkwindow.c:9759
#75 gtk_window_check_resize (container=0x5555559062e0) at gtkwindow.c:8516
#76 0x00007ffff73c10b6 in _g_closure_invoke_va (closure=0x5555556748b0, return_value=0x0, instance=0x5555559062e0, args=0x7fffffffd040, n_params=0, param_types=0x0)
    at ../gobject/gclosure.c:873
#77 0x00007ffff73de330 in g_signal_emit_valist (instance=0x5555559062e0, signal_id=<optimized out>, detail=<optimized out>, var_args=var_args@entry=0x7fffffffd040)
    at ../gobject/gsignal.c:3300
#78 0x00007ffff73de933 in g_signal_emit (instance=<optimized out>, signal_id=<optimized out>, detail=<optimized out>) at ../gobject/gsignal.c:3447
#79 0x00007ffff7848058 in gtk_container_idle_sizer (clock=0x55555568e2d0, container=0x5555559062e0) at gtkcontainer.c:2064
#80 0x00007ffff73c0e5d in g_closure_invoke (closure=0x7fffdc02cd20, return_value=0x0, n_param_values=1, param_values=0x7fffffffd300, invocation_hint=0x7fffffffd280)
    at ../gobject/gclosure.c:810
#81 0x00007ffff73d4f23 in signal_emit_unlocked_R (node=node@entry=0x5555556691e0, detail=detail@entry=0, instance=instance@entry=0x55555568e2d0, 
    emission_return=emission_return@entry=0x0, instance_and_params=instance_and_params@entry=0x7fffffffd300) at ../gobject/gsignal.c:3635
--Type <RET> for more, q to quit, c to continue without paging--
#82 0x00007ffff73de2da in g_signal_emit_valist (instance=<optimized out>, signal_id=<optimized out>, detail=<optimized out>, var_args=var_args@entry=0x7fffffffd4b0)
    at ../gobject/gsignal.c:3391
#83 0x00007ffff73de933 in g_signal_emit (instance=instance@entry=0x55555568e2d0, signal_id=<optimized out>, detail=detail@entry=0) at ../gobject/gsignal.c:3447
#84 0x00007ffff7611e73 in _gdk_frame_clock_emit_layout (frame_clock=frame_clock@entry=0x55555568e2d0) at gdkframeclock.c:634
#85 0x00007ffff761256a in gdk_frame_clock_paint_idle (data=0x55555568e2d0) at gdkframeclockidle.c:437
#86 0x00007ffff75fcc8c in gdk_threads_dispatch (data=0x555555802000, data@entry=<error reading variable: value has been optimized out>) at gdk.c:768
#87 0x00007ffff72de007 in g_timeout_dispatch (source=0x7fffdc02cda0, callback=<optimized out>, user_data=<optimized out>) at ../glib/gmain.c:4667
#88 0x00007ffff72dd528 in g_main_dispatch (context=0x5555555e19b0) at ../glib/gmain.c:3182
#89 g_main_context_dispatch (context=context@entry=0x5555555e19b0) at ../glib/gmain.c:3847
#90 0x00007ffff72dd918 in g_main_context_iterate (context=context@entry=0x5555555e19b0, block=block@entry=1, dispatch=dispatch@entry=1, self=<optimized out>)
    at ../glib/gmain.c:3920
#91 0x00007ffff72dd9b0 in g_main_context_iteration (context=context@entry=0x5555555e19b0, may_block=may_block@entry=1) at ../glib/gmain.c:3981
#92 0x00007ffff74da6b5 in g_application_run (application=0x5555555db0e0, argc=<optimized out>, argv=0x7fffffffd8c8) at ../gio/gapplication.c:2470
#93 0x000055555556104d in _vala_main (args=0x7fffffffd8c8, args_length1=1) at /home/alberto/projects/vala/sequeler/src/Main.vala:31
#94 0x0000555555561090 in main (argc=1, argv=0x7fffffffd8c8) at /home/alberto/projects/vala/sequeler/src/Main.vala:25
```
</details>

which seems related to css?

### CODE REVIEW

**Comment by albfan** [comment]
After rebase on top of last dev commits, cannot reproduce that last bt, but other related with ssh tunneling. Here after try several test connections:

<details>
<summary>bt</summary>

```
Thread 7 "com.github.alec" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffe6b11700 (LWP 247)]
0x00007ffff6fd7400 in _libssh2_transport_send (session=session@entry=0x7fffd801e4f0, data=data@entry=0x7fffd802b308 "\001", data_len=<optimized out>, 
    data2=data2@entry=0x5555555ae8e9 "", data2_len=<optimized out>) at src/transport.c:844
844	src/transport.c: No such file or directory.
(gdb) bt
#0  0x00007ffff6fd7400 in _libssh2_transport_send (session=session@entry=0x7fffd801e4f0, data=data@entry=0x7fffd802b308 "\001", data_len=<optimized out>, 
    data2=data2@entry=0x5555555ae8e9 "", data2_len=<optimized out>) at src/transport.c:844
#1  0x00007ffff6fd160f in session_disconnect (lang=0x5555555ae8e9 "", description=0x5555555af4b3 "Client disconnecting normally", reason=11, session=0x7fffd801e4f0)
    at src/session.c:1130
#2  libssh2_session_disconnect_ex (session=0x7fffd801e4f0, reason=11, desc=0x5555555af4b3 "Client disconnecting normally", lang=0x5555555ae8e9 "") at src/session.c:1150
#3  0x0000555555592b5d in sequeler_services_connection_manager_ssh_tunnel_close (self=0x555555d7b6d0, from=0x7fffd802ea40 "ConnectionManager.vala:276")
    at /home/alberto/projects/vala/sequeler/src/Services/ConnectionManager.vala:385
#4  0x00005555555915e7 in sequeler_services_connection_manager_ssh_tunnel_open (self=0x555555d7b6d0, is_real=0, error=0x7fffe6b10c28)
    at /home/alberto/projects/vala/sequeler/src/Services/ConnectionManager.vala:276
#5  0x000055555558f297 in sequeler_services_connection_manager_ssh_tunnel_init (self=0x555555d7b6d0, is_real=0, error=0x7fffe6b10c88)
    at /home/alberto/projects/vala/sequeler/src/Services/ConnectionManager.vala:132
#6  0x00005555555a2b92 in __lambda66_ (_data34_=0x555555f39d00) at /home/alberto/projects/vala/sequeler/src/Widgets/ConnectionDialog.vala:587
#7  0x00005555555a2d15 in ___lambda66__gthread_func (self=0x555555f39d00) at ConnectionDialog.c:4216
#8  0x00007ffff7306879 in g_thread_proxy (data=0x5555559dead0) at ../glib/gthread.c:784
#9  0x00007ffff6e99f7b in start_thread (arg=<optimized out>) at pthread_create.c:463
#10 0x00007ffff6dcb3af in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
(gdb) 

```
</details>

**Comment by Alecaddd** [comment]
Yeah, I updated a lot of methods to be wrapped around `Threads` and use `idle` to update the UI, that fixed a lot of issues for me, and now I can connect via SSH even when double-clicking on a saved connecting in the library without opening the Dialog.

Thank you so much for all your commits, I'm gonna write some comments on them mostly to understand what's happening before merging.

**Comment by albfan** [comment]
Nice to hear. Have to give it a try. I found that SIGSEV on ssh was because when ssh_tunnel_close is called, sometimes that makes existing tunnel to end calling direct_shutdown, where internal vars are not avaliable anymore, as self->priv->ssh_tunnel_alive or similar were freed before.

About tunnels kept open, one case is: testing a connection, it ends tunnel correctly as remaining len to read is 0, but ssh_tunnel_close is never called. Next tries to call test or connect fails as there's an existing tunnel open.

Looking forward for those new commits!

File: src/Application.vala [hunk 1]
```diff
@@ -90,6 +90,8 @@ public class Sequeler.Application : Gtk.Application {
 
     protected override void activate () {
         var window = new Sequeler.Window (this);
+        Gtk.Settings.get_default().set_property("gtk-theme-name", "elementary");
```
Comment (bilelmoussaoui) [inline src/Application.vala]: Why those are enforced here?

Comment (bilelmoussaoui) [reply src/Application.vala]: This might causes issues if the user doesn't have the elementary gtk/icons theme installed.

Comment (albfan) [reply src/Application.vala]: You're right, but without it, `office-database` icon (on ConnectionDialog) is not resolved.

Not sure how to deal with sequeler build outside elementary theme

**Comment by Alecaddd** [comment]
Could you please split up this PR into multiple PRs and hold back all the flatpak related updates?
It's easier to track changes, thanks.

**Comment by albfan** [comment]
@Alecaddd sure! let me rework it

**Pull request closed without merging**